lua/heirline/statusline.lua

```diff
@@ -272,32 +272,56 @@ local function register_global_function(component)
     return "v:lua." .. func_name
 end
 
+---@param data table
+local function create_event(data)
```

Comment:
This comment is optional, but something like `create_autocmd_args` might be clearer to me than "event".

`event` is a string like `"ModeChanged"`, whereas these are basically arguments used to create an auto-command.